src/pyhf/paramsets.py

```diff
@@ -35,6 +35,10 @@ def __init__(self, **kwargs):
 
     def expected_data(self, pars):
         tensorlib, _ = get_backend()
+        sh = tensorlib.shape(pars)
```

Comment:
this doesn't make much sense. Why are the `pars` multiplied by the `fc`?

Reply:
why does it not make sense? this is just a generalization of the line below to the batched case.

Reply:
I think this can be replaced with an einsum, unless I'm mistaken... What's the return shape of `expected_data` for `pars = (B, npars)`?

Reply:
i don't think `einsum` makes this more readable. The return shape is the same as `pars`

Reply:
```suggestion
        sh = tensorlib.shape(pars)
        # if batched, tile the factors as they only depend on background uncertainty
```